Do not consider scrutinees of projections and cases as heads for hint mode "!" (#14392)
coq/coq: doc/changelog/02-specification-language/14392-nohead-evar-application-hea.rst

- **Changed:**

>   The mode ``!`` of hints matches a term iff the applicative head is not an existential variable,
>   now matching also a projection applied to any term or a `match` on any term.

Reviewer: ```suggestion
  The hints mode ``!`` matches a term iff the applicative head is not an existential variable.
  It now also matches projections applied to any term or a `match` on any term.
```
